samples/fetch_data/src/sample/fetch_data.cljd

```diff
@@ -0,0 +1,45 @@
+(ns sample.fetch-data
+  (:require ["package:flutter/material.dart" :as m]
+            ["package:http/http.dart" :as http]
+            ["dart:convert" :as convert]
+            ["dart:async"]
+            [cljd.flutter.alpha :as f]))
+
+
+(defn decode-album
+  [raw-album]
+  (let [json-album (convert/jsonDecode raw-album)]
+    (dart:core/print (get json-album "title"))
+    {:user-id (get json-album "userId") :id (get json-album "id") :title (get json-album "title")}))
+
+
+(defn fetch-album
+  []
+  (let [response (await 
+                  (http/get
+                   (dart:core.Uri/parse "https://jsonplaceholder.typicode.com/albums/1")))]
+     (cond (= (.statusCode response) 200) (decode-album (.body response)) :else {})))
+
+(def my-app
+  (f/widget
+   :state [state {:future-album (fetch-album)}] 
+   (m/MaterialApp
+    :title "Fetch Data Example"
+    :theme (m/ThemeData :primarySwatch m.Colors/blue)
+    :home
+    (m/Scaffold
+     :appBar (m/AppBar :title (m/Text "Fetch Data Example"))
+     :body
+     (m/Center
+      :child 
+      (m/FutureBuilder 
+       :future (:future-album @state)
```

Comment:
why not just put `(fetch-album)` here and get rid of the state?
Or use a plain-old-let

Reply:
If I use `let`, it will be on which level?

Reply:
On the line you have the `:state` instead put `(let [future-album (fetch-album)] ...)` and on line 36 `:future future-album`